test/membrane_hackney/source_test.exs

```diff
@@ -38,24 +54,18 @@ defmodule Membrane.Hackney.SourceTest do
         async_response: :mock_response
       })
 
-    [state_streaming: state]
-  end
-
-  test "handle_playing_to_prepared/2 should close request when moving from :playing to :prepared" do
-    state = %{@default_state | async_response: :mock_response}
-    mock(:hackney, close: 1)
-    assert {:ok, new_state} = @module.handle_playing_to_prepared(nil, state)
-    assert new_state.async_response == nil
-    assert_called(:hackney, :close, [:mock_response])
```

Comment:
I believe we need to have a test that would check if the `:hackney.close` is always called (as far as I can see, the rest of the tests checks only if the resource is cleaned once an error occurs)

Reply:
It is checked in lines 312-319